### PR TITLE
New biopython logo

### PR DESCRIPTION
**Pull request by padix-key**
This is a suggestion for a new Biopython logo. The image is an vector graphic created with Inkscape.

### CODE REVIEW

**Comment by peterjc** [comment]
You'd best raise this on the Biopython mailing list http://biopython.org/wiki/Mailing_lists - Personally I much prefer the current logo, although a (vector?) version of that suitable for use at small sizes might be nice...

We'd also request you explicitly agree to make this available under both the _Biopython License Agreement_ AND the _BSD 3-Clause License_ (in line with efforts to dual-license the whole project).

**Comment by padix-key** [comment]
Okay, thank you.

I aggree that the committed content is available under both, the Biopython License Agreement AND the BSD 3-Clause License.

**Comment by peterjc** [comment]
Thanks!

**Comment by bow** [comment]
Thanks for the suggestions @padix-key ~ I actually like this better than the current logo. Is it an option to use another font? For example, something like one of these: [1](https://fonts.google.com/specimen/Comfortaa), [2](https://fonts.google.com/specimen/Sarala), [3](https://fonts.google.com/specimen/Alegreya+Sans+SC) ~ which seems more sleeker for logos than Calibri.

**Comment by padix-key** [comment]
Yes, that is totally possible. I really like the Comfortaa font you suggested. I will update the commit with images using this font.

**Comment by MarkusPiotrowski** [comment]
Can you also try to make an icon out of it, which could probably used as favicon for the webpage?

**Comment by peterjc** [comment]
Good point about the favicon - the current logo is unsuitable for that and not idea at the scale used in Twitter profile images either.

**Comment by padix-key** [comment]
The images are now updated.

**Comment by gedankenstuecke** [comment]
Personally I have to say that I find this suggestion much friendlier and more inviting, for a lack of better words, than the old one. It also performs much better at different sizes than the old one.

**Comment by kescobo** [comment]
This is pure pedantry on my part - but it should be possible to make the [major/minor grooves](http://atlasgeneticsoncology.org/Educ/Images/DoublehelixE.jpg) a bit more obvious. Could just scoot the yellow several pixels to the right.

I bring this up because I currently work at a place where I'm surrounded by medical illustrators, and they all say their biggest pet peeve (other than right handed DNA) is lack of proper proportions in the major/minor groove.

File: _includes/sidebar.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
       -->
       <div class="sidebar-logo">
         <a href="{{ site.baseurl }}">
-          <img src="{{ site.baseurl }}assets/images/biopython-transparent.png" alt="Biopython Logo" title="Biopython Home" width="100%"/>
+          <img src="{{ site.baseurl }}assets/images/biopython_logo_medium.png" alt="Biopython Logo" title="Biopython Home" width="100%"/>
```
Comment (peterjc) [inline _includes/sidebar.html]: I don't see a new file by this name: ``biopython_logo_medium.png``

**Comment by peterjc** [comment]
The blue and yellow colours are clearly playing to the current official Python logo, which I note is apparently also described as two snakes https://www.python.org/community/logos/

If the Biopython community consensus is to try to adopt something like this as our logo, we would probably have to officially ask the PSF for permission given the potential trademark concerns.

**Comment by peterjc** [comment]
Should the text be "Biopython" (capital B only, not a capital P), in line with the historic policy in our FAQ?

**Comment by peterjc** [comment]
Some positive comments on the mailing list thread: http://mailman.open-bio.org/pipermail/biopython/2017-May/016157.html

**Comment by MarkusPiotrowski** [comment]
@peterjc I like the lowercase logo. Python also uses a lowercase 'python' in its logo.
@padix-key I'm splitted about the updated font. It looks quite cool but it makes the logo even more similar to Python's original logo.

**Comment by peterjc** [comment]
Font similarity to the original logo can be a plus - especially if we try to have both (e.g. current logo may still work nicely in backgrounds or masthead).

I would like something which works well square for smaller things like Twitter & GitHub avatars - the yellow-blue-snake-helix above the word biopython ought to work much better here.

**Comment by liamhawkins** [comment]
I really like this suggested logo. To my eyes I'd like to see the thickness of the snakes increased by 5-10%, especially for the 'icon' image, any thoughts?

**Comment by peterjc** [comment]
On the mailing list c0d3g33k wrote:
>I might suggest a little tweaking of the head area, since it looks more planarian than python.  
>
> Otherwise, it gets my vote.

I'd also like the snakes to have a bit more 'personality' somehow, eyes or tongue maybe? Or do they need a little light/dark or hints of scales?

**Comment by kescobo** [comment]
@peterjc I have some compile time so I took a quick and dirty stab at adding scales - 

screen shot 2017-05-08 at 1 09 42 pm

I think I prefer the no-scales version, but if folks like where this is going, I can work on it a bit more.

**Comment by padix-key** [comment]
Thank you for your feedback. The heads are a little bit more "snaky" now. I also tested the "biopython" text with capital "B", for the reason @peterjc mentioned, but I think the lower case "b" looks better. @peterjc and @MarkusPiotrowski : You are right, the usage of two snakes are quite similar to the "Python" logo, but other projects use the snake reference als well (PyDev, SciPy, Spyder etc.). So I think this is not a trademark problem, but I am not an expert.

**Comment by peterjc** [comment]
There's been some feedback in parallel on the mailing list including another suggestion to try a tongue for the snake heads. Almost the original logo competition entries had red tongues - http://biopython.org/wiki/Logo

**Comment by peterjc** [comment]
I'm happy to accept "biopython" rather than "Biopython" in the logo for stylistic reasons - after all the current logo uses "BIOPYTHON" in all capitals.

**Comment by gedankenstuecke** [comment]
I think the lastest draft is much better. And I agree with @kescobo that the scales don't look ideal. It also removes a lot from the great scalability, which is a big improvement over the current logo! :+1:

**Comment by peterjc** [comment]
@kescobo thanks for trying that scales idea - sadly I agree plain works better than this.

Henrik Vestergaard and Thomas Hamelryck managed to do a 3D effect very nicely with the lighting on the old logo.

Here in my mind I was picturing was some way to distinguish the under belly and thus visually convey the snake is twisted in 3D space around the axis of the helix. Perhaps two shades of blue/yellow would work for showing the underbelly? I've tried and failed to find any line art showing what I have in mind - http://www.dragoart.com/tuts/21435/1/1/how-to-draw-ouroboros.htm is the closest - even just small black lines indicating the belly scales might be enough...

**Comment by peterjc** [comment]
I've sketched what I had in mind for hinting at the belly scales - these could just be thin black lines only visible at larger sizes? Another thin line for the backbone might help too when viewing the image at larger sizes.



(This is only red and blue hatched due to not having anything more suitable to hand to mimic the current uniform yellow and blue fill).

The snake head and tail position in my sketch is based off the current logo, although the twisting of the snakes has come out a little differently.

**Comment by padix-key** [comment]
@peterjc Maybe the belly scales could fit into the new logo, but I am am definetly not skilled enough to make this look properly.

**Comment by peterjc** [comment]
Oh well, the belly scales would have to fade away / vanish for small versions of the logo anyway.

I can see advantages in deliberately going for an abstract snake. For example the way the left hand snake head looks on the current logo bothers me, as it seems the snake is holding its head upside down.

**Comment by gedankenstuecke** [comment]
Maybe the "flat" design would look a bit better if one would go for the "side"-view of the snakes as @peterjc did draw them in his view, instead for the "top-down" view as in the current design (that seems to be the approach the python logo itself takes as well)?

**Comment by carlosmr12** [comment]
What a great work! Well done!

**Comment by padix-key** [comment]
I created a logo with side view head. Which one do you like better in principle?
biopython_logo

**Comment by bow** [comment]
Thank you for putting in the work @padix-key!

If I have to pick, I would go for the ones with the eye both visible (so from the top). I have a preference towards more minimalist style (so two eyes on the top, without any scales or shadow is perfect for me).

**Comment by padix-key** [comment]
I also prefer the top view.

**Comment by gedankenstuecke** [comment]
Personally I find the side view more appealing, but I'd flip the yellow snake's head, so it's not standing on it's head. But really, both versions are great 👍

**Comment by peterjc** [comment]
Presumably the PNG images are simply exported from Inkscape at different resolutions? So it should be easy to make any specific size needed as needed?

**Comment by padix-key** [comment]
Yes

**Comment by peterjc** [comment]
Like Bastien said, for the side-view version I'd also flip the yellow snake's head so it is right side up (even though this breaks the symmetry).

How do you think having the necks turned to have the heads more level would look (i.e. a bit more like the 2003 logo heads)?

**Comment by padix-key** [comment]
@gedankenstuecke and @peterjc If the consensus is the side view, I will create a version with a flipped yellow head. @peterjc I am not sure, if the turned necks would fit well into the sine form of the logo.

**Comment by peterjc** [comment]
(Don't worry about it for now, but in some of the PNG images the three junctions where the blue snake bends are visible as grey/white lines)

**Comment by MarkusPiotrowski** [comment]
Just in case my thumb-up to Bow's comment isn't realized: I like the previous version (top view) more.

**Comment by padix-key** [comment]
@peterjc The thing with the white lines is a nasty problem, that happens in the conversion from svg to png at some resolutions. In later use at the website, GitHub, Twitter we can use resolutions, where this effect is not happening.

**Comment by peterjc** [comment]
On the mailing list Matthieu Bruneaux  made this blunt-nose version, which seems popular (and I agree its an improvement for the over-head view of the snake heads):

biopython_logo_xs_short-snout

Also on the mailing list, based on that Jocelyne made this two-tone version:

biopython_logo_xs_short-snout_color

I'd like to see this will little red tongues on the snakes 😛

**Comment by peterjc** [comment]
@kescobo does the major/minor grove look clear enough to you in the current logo versions on this pull request, or does it need slightly more emphasis by moving the yellow snake further to the right?

**Comment by etal** [comment]
:+1: for the blunt-nose design. No tongues -- it's cleaner.

The two-tone shading would work well for larger versions of the design, like the front page of the Tutorial and Cookbook, while flat color works better for smaller images and icons.

Last year, BioJava used a GitHub issue (biojava/logo#10) to conduct a runoff vote on several candidate versions of their new logo, with each design posted as a separate comment and each dev/user voting with a +1 on the preferred design(s).

**Comment by peterjc** [comment]
Good point @etal - I was wondering if we might want to do something similar for voting, with an issue comment for each shortlisted design variant shown at several sizes, say:

* favicon (no text),
* avatar (e.g. GitHub/Twitter - no text?),
* medium (e.g. website side bar)
* large (e.g. cover of Tutorial, Twitter Banner)

However it seems we may be converging on a consensus, so it might just be a two (three?) way race between the old 2003 logo and this potential replacement (and any middle ground compromise like the new logo used at smaller sizes, but the old logo persisting as a banner image in places?).

**Comment by kescobo** [comment]
@peterjc I consulted with my design-minded colleagues. Consensus was that it looks like it needs a bit of extra spacing. It actually is also a bit squished - I grabbed an image of DNA ([from here](https://en.wikipedia.org/wiki/File:DNA_Structure%2BKey%2BLabelled.pn_NoBB.png)) and used it to map the snakes onto it:

1artboard 1

Which looks like this:

2artboard 1

Note - I based this off current svg, so there's no flat head (I also like that) and I tried another shading option.

**Comment by MarkusPiotrowski** [comment]
I wonder if we push it a little bit far with a nanometer exact positioning of the two snakes (and many of the other many little suggestions). Already the first proposal from @padix-key was very cool, modern, and far better than the old/actual logo. The old logo had its time and I wonder if, at the time it was elected, there were so many demands put on it.

**Comment by bow** [comment]
I tend to concur with @MarkusPiotrowski. If I may say, perhaps having the images at two aspect ratios are enough:

1. one for the rectangular form with the 'biopython' word on it
2. one in as a square without any words

Also, I propose to limit the choices to three:

1. One that @padix-key key thinks is best (almost all suggestions are based on his initial design after all).
2. One for the current consensus (with the shadowing, side-view heads, etc.).
3. The old logo.

That's already 5 or 6 mockups (I believe the old logo does not have a square version?). Maybe even limit it to just the rectangular aspect ratio? That puts us at 3 :).

**Comment by peterjc** [comment]
Most of the mock ups are people trying out ideas or visually explaining their suggestions. @padix-key gets to ignore these using his artistic judgement. I hope we converge on one candidate replacement logo which can then be voted on.

**Comment by padix-key** [comment]
I uploaded a new version, with a flat nose and a tongue. A colr gradient, which looks good is quite diffucult to include into the vector graphic, but I tried something else with these light and dark segments (Do you think they look good?). The issue with those thin white lines @peterjc montoined is fixed now.
@kescobo I am a fan of a more compact design, even if it's less realistic. And who says that this is DNA, I just see two intertwined snakes ;)

**Comment by gedankenstuecke** [comment]
That's great, I'm now converted to liking the flat, top-down heads much more as well! 👍 💖

**Comment by bow** [comment]
Looking great @padix-key ~ the latest iteration without any gradients would be perfect I think. And I have to say I think I like it better now with the tongues, which was not the case. Well done!

**Comment by peterjc** [comment]
I like the new snake heads with the blunt nose and tongue 👍 

I prefer the single colours to this light/dark segment idea. I can understand using light/dark to suggest twisting being quite tricky to implement in vector art, so it make sense to drop that idea.

Regarding the helix placement, I think right now it looks good. Even if this doesn't fully match the real DNA structure, there is a recognisable deliberate major/minor grove.

**Comment by widdowquinn** [comment]
Just to repeat comments from the mailing list for completeness - I much prefer the single-colour snake versions (which I think will render better for greyscale, too), and the version without snake tongues (which I think will not render visibly in smaller scaling of the logo).

My view is that the simpler design is more elegant, and will render more consistently across a wider range of scales and applications.

**Comment by kescobo** [comment]
@padix-key @peterjc Agreed - I actually don't think that making it match perfectly is necessary or in this case desirable. Latest linked version looks good to me re: major/minor grooves.

If you're going to do two-tone, I think the darker segments should be the ones that pass behind. I also prefer the non-gradient versions, but that said, vector gradients aren't actually that hard.

**Comment by jocelyne8** [comment]
For 2 tones, I think belly / back looks better. I tried editing the svg but my skills suck...
![biopython_logo]
(https://cloud.githubusercontent.com/assets/14894193/25905296/7cab519e-3556-11e7-98dc-f5069e6b547d.png)

**Comment by peterjc** [comment]
Thanks @jocelyne8 - that two tone prototype is useful, although where to put the shading to best emphasise twisting is tricker.  Still, I've come round to agreeing single colour blue and yellow snakes is the more elegant choice.

**Comment by jocelyne8** [comment]
I agree. I prefer either a front / back 2-tone or 1-tone. The segmented 2 tone looks, well, segmented ;)

**Comment by cbrueffer** [comment]
Lots of great logo ideas! Regarding shapes I like @padix-key's latest top view flat nose + tongue version, however I prefer the minimalistic colors of the original submission.

**Comment by padix-key** [comment]
Thank you all for your constructive feedback. I think it is time to vote for the final version of the logo. I will post 4 logo suggestions including the old logo. A thumbs up for the the respective post means a vote for this logo.
Furthermore I will post an example logo without tongue. A thumbs up for this post means a vote against tongue, a thumbs down means a vote for tongue.
So eventually everyone has two votes: one for the overall arrangement and one for or against the tongue.
I know that this system only reaches the people on GitHub, but I think most contributors are on GitHub anyway. You will be able to vote for two days from now on. After that, I will commit the final logo and the pull request can be merged. I hope this procedure is OK for everyone.

**Comment by padix-key** [comment]
Option 1: The old logo
biopython_logo_old

**Comment by padix-key** [comment]
Option 2: The vanilla logo
biopython_logo_vanilla

**Comment by padix-key** [comment]
Option 3: The segmented logo
biopython_logo_seg

**Comment by padix-key** [comment]
Option 4: The shaded logo (thanks to @jocelyne8):
biopython_logo_shadow

**Comment by padix-key** [comment]
An the decision for or against the tongue. (Thumbs up means vote against tongue)
biopython_logo_notongue

**Comment by padix-key** [comment]
Thank you for voting. The final logo has been uploaded, including a small version without text as icon, so this pull request should be ready to be merged. Should we update or remove "wiki/Logo.md"?

**Comment by peterjc** [comment]
It does look like a quite clear preference from the voting - the single colour blue-yellow snakes (no shading), and without the little tongues.

@padix-key I would recommend updating ``wiki/Logo.md`` rather than deleting it.

I think next we should email the PSF just to make sure they have no trademark issues with the proposed logo - I'll draft something and CC Patrick and the mailing list on this.

**Comment by padix-key** [comment]
Logo.md is also updated now, so when the PSF gives its OK, the request can be merged.

**Comment by bede** [comment]
@padix-key Thank you for contributing this new logo – it was much needed. May I suggest that you convert the logotype to vector paths so that they are not dependent on the client's font renderer. I just downloaded the SVG and was greeted with Times New Roman logotype since I do not have Comfortaa installed.

**Comment by padix-key** [comment]
Good point, I fixed that

**Comment by peterjc** [comment]
I've written a draft email to the PSF about the logo for @padix-key and the rest of the OBF Board to comment on, CC'ing Iddo as a former Biopython project lead as he raised the point about licensing on the mailing list:

@idoerg wrote:

> ..., if we adopt, I think we should license the logo appropriately. See
> the Tux logo as an example: https://commons.wikimedia.org/wiki/File:Tux.svg

http://mailman.open-bio.org/pipermail/biopython/2017-May/016171.html

**Comment by peterjc** [comment]
As per https://www.python.org/psf/trademarks/ email sent to PSF about the proposed new Biopython logo, asking if the regard the proposed new Biopython logo as a derived from the Python logo, and if so would the PSF grant us permission to use it?

**Comment by peterjc** [comment]
Does anyone have strong views about how the new logo ought to be licensed?

I would be happy with dual licensing under both the Biopython License Agreement AND the BSD 3-Clause License (in line with efforts to dual-license the whole project).

However, for some purposes a software license is perhaps not ideal, so OBF President @hlapp suggested using CC0 or CC-BY. This would need some thought/care for how to word our README/LICENSE files for distributing the logo with the Biopython releases (e.g. as part of the documentation).

**Comment by hlapp** [comment]
To clarify, one of the main issues with trying to apply an open-source license to artwork is that much of the language in an open-source license is about _source code_, and in fact what is being licenses _is_ the source code. (See for example [[1]].)

In the case at hand, with the format of the artwork in question being SVG, one could actually argue that the open-source license language being about and pertaining to source code would have a clear and obvious interpretation also for the artwork. So maybe in this concrete case, the issues one would normally run into with using an open-source software license for artwork would be largely or entirely mitigated by the choice of format for the artwork.

[[1]]: https://www.linux.com/news/trouble-artwork-and-free-software-licenses

**Comment by peterjc** [comment]
We got a very nice reply from David Mertz at the PSF, which he is happy to be shared:

> ---------- Forwarded message ----------
> From: David Mertz
> Date: Tue, May 23, 2017 at 9:02 PM
> Subject: Re: [psf-trademarks] Proposed logo usage enquiry from Biopython to PSF
> To: Peter Cock <p.j.a.cock@googlemail.com>
> Cc: psf-trademarks, Patrick Kunzmann, OBF Board
>
>
> Dear Peter,
> 
> We have discussed your logo, and have determined that it falls in the "inspired by but not derived from" the Python logo category.  Therefore, this usage is outside the scope of our copyright and we have no concern about its use.
> 
> Speaking just personally, FWIW, I think the new design is very attractive, and wish you and the project the best success (I've used and followed the Biopython project myself).
>
> Best wishes, David Mertz
> PSF Trademarks Committee

**Comment by peterjc** [comment]
Just sent to the biopython and biopython-dev mailing lists:

http://mailman.open-bio.org/pipermail/biopython-dev/2017-May/021711.html
http://mailman.open-bio.org/pipermail/biopython/2017-May/016242.html

> Hello all,
>
> To recap for those not following the GitHub issue [1], based on
the voting there was a clear preference for a monotone yellow and
blue version of Patrick's logo using an overhead view of the snake
heads with blunt nose and no tongue.
> 
> We have confirmed with the Python Software Foundation (PSF)
that they have no trademark concerns since the current Python
logo is also a yellow and blue snake [2].
>
> If there are no objections here on the mailing list or on the GitHub
issue, I propose we adopt the new logo at the end of next week
(i.e. Friday 9 June).
>
> As a curtesy, I will forward this email directly to past major
contributors including the named authors of the 2009 publication,
in case they have missed these discussions.
>
> Thank you all - especially Patrick for coming up with a stylish fresh
new take on our classic logo (and for getting involved on the code
side of Biopython as well).
>
> Regards,
> 
> Peter
> 
> [1] https://github.com/biopython/biopython.github.io/pull/116
> [2] http://mailman.open-bio.org/pipermail/biopython-dev/2017-May/021703.html
> and https://github.com/biopython/biopython.github.io/pull/116#issuecomment-303666735

**Comment by peterjc** [comment]
And another email this time including the low volume announcements mailing list:
* http://mailman.open-bio.org/pipermail/biopython-announce/2017-May/000171.html
* http://mailman.open-bio.org/pipermail/biopython/2017-May/016243.html
* http://mailman.open-bio.org/pipermail/biopython-dev/2017-May/021712.html

**Comment by peterjc** [comment]
I've sent an email to mailing lists announcing the change, and will add links here once mailman has updated:

> Dear Biopythoneers,
>
> No one has raised any objections, so after 14 years of usage we will transition away from the DNA structure inspired two yellow snake double helix logo by  Henrik Vestergaard and Thomas Hamelryck - and adopt the new blue and yellow snake double helix logo contributed by Patrick Kunzmann.
>
> http://biopython.org/wiki/Logo
> https://github.com/biopython/biopython.github.io/pull/116
>
> I will always have a soft spot for the old logo, which was introduced shortly before I first started using Biopython, but am looking forward to seeing the new logo being adopted.
>
> Thank you Henrik, Thomas, and Patrick,
> 
> Kind regards,
>
> Peter

**Comment by peterjc** [comment]
Congratulations and thank you @padix-key - the new logo is live on http://biopython.org/ and https://twitter.com/Biopython now 👍 

Could you make a similar pull request on the main repository where we use the logo in the Tutorial please?

I am inclined to keep https://github.com/biopython/biopython/blob/master/Doc/images/biopython.jpg in place, and use it and the new logo (e.g. ``biopython_blue_yellow.png`` or ``biopython_2017.png``) for a new FAQ entry in the Tutorial. Or we could just replace the old PNG image and add a PDF?

Ideally could you provide a (transparent background?) PNG and PDF (with the same base filename) which would be ideal for us to generate the Tutorial in HTML and PDF format.

You can add yourself to the ``CONTRIB.rst`` and ``NEWS.rst`` file for the next release, and I think we should mention the new logo in the notes for upcoming Biopython 1.70 too :)

**Comment by peterjc** [comment]
For now I have remove the Twitter header banner - trying to use the snakes part of the logo didn't look great somehow (I tried a grey background), and in TweetBot on the Mac it looked weird.

**Comment by peterjc** [comment]
See https://github.com/biopython/biopython/pull/1285 for applying the logo to the main repository and using it in the Tutorial and README file etc.

**Pull request merged**